test/fail_compilation/b6226.d

```diff
@@ -0,0 +1,22 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b6226.d(13): Error: cannot implicitly convert expression `400000000` of type `int` to `ubyte`
+fail_compilation/b6226.d(19): Error: cannot implicitly convert expression `200` of type `int` to `byte`
```

Comment:
This message is also wrong. 200 is not converted to a byte, x is converted to an int.

Reply:
200 is really converted to a byte. Promotion of the condition is moved after semantic check of the case exactly so that implicitCast use the original condition type as target of the implicit conv.